### PR TITLE
27-28-w03-d01-express-params-queries

### PR DESCRIPTION
**Pull request by tgaff**
Frustratingly #66 was closed by merging in the material from 22-25 on a separate PR.  This naturally makes it look like we're not working on this and hides any discussion.  

Lesson learned:

Either 1) use an issue instead of an empty branch + .gitkeep    OR   2) make sure you create 27-28 directory in your branch with a .gitkeep and a separate commit that won't be merged by the PR for previous-cohorts.  :angry:

### CODE REVIEW

File: 02-express-mongo-crud/express-params-queries/27-28/README.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+Common cases for Query Params:
+* searches             -  GET /?q=kittens
+* optional selections  -  GET /calculator?lang=en
+* pagination           -  GET /posts?start=1&size=10
+* other limits         -  GET /posts?since=2015-11-29
+
+Finally you might use them both together in some cases: `/posts/33/comments?limit=50`
+
+## Middleware
```
Comment (tgaff) [inline 02-express-mongo-crud/express-params-queries/27-28/README.md]: So in 25 we did middleware here very briefly.  However if we're doing body-parser we have to first talk about POST and that really would need it's own section.  27 hasn't seen POST used yet.   @justincastilla  should we just take out body-parser?

File: 02-express-mongo-crud/express-params-queries/27-28/README.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+Common cases for **Query Params**:
+* **searches**             -  GET `/?q=kittens`
+* **optional selections**  -  GET `/calculator?lang=en`
+* **pagination**           -  GET `/posts?start=1&size=10`
+* **other limits**         -  GET `/posts?since=2015-11-29`
+
+Finally you might use them both together in some cases: `/posts/33/comments?limit=50`
+
+## Middleware
```
Comment (tgaff) [inline 02-express-mongo-crud/express-params-queries/27-28/README.md]: So in 25 we did middleware here very briefly. However if we're doing body-parser we have to first talk about POST and that really would need it's own section. 27 hasn't seen POST used yet. @justincastilla should we just take out body-parser?

Comment (justincastilla) [inline 02-express-mongo-crud/express-params-queries/27-28/README.md]: I think body-parser is a nice to have but module 2 of the first day might be a little aggressive.  We had the luxury of having very few students where we could hand-guide them; middleware is a bit harder to explain without closer supervision.  I think students would benefit more from a thorough overview of POST.  We could include adding and using body-parser as part of hw/reading.  The two different ways of using body-parser ( app.use() and parseUrlEncoded) will really confuse them if we try to cram too much in one day.  Also creating your own middleware section will hurt.  

I say we keep middleware documentation on a separate md file.  IF we feel the students are at a good level after we cover query/params/POST/GET/routes we should introduce them verbally to middleware.  If not, we can assign it to them as hw/reading.

Comment (tgaff) [inline 02-express-mongo-crud/express-params-queries/27-28/README.md]: I agree, you're absolutely right.  They never really GOT middleware last time anyway.  

Do you want me to write-up POST and move middleware or do you want to take that on?

**Comment by tgaff** [comment]
FYI I started work on #75 if you want to help out.

**Comment by justincastilla** [comment]
I can add some POST to the main document. I'll have that by 530

**Comment by tgaff** [comment]
OK, I'm pushing some stuff in #75; feel free to change that as you see fit.

**Pull request merged**